basics/uattributes.adoc

```diff
@@ -59,29 +59,57 @@ The following table defines attributes that are used for all message types:
 |`id`
 |link:uuid.adoc[UUID]
 |yes
-|A unique message identifier. The timestamp contained in the UUID is used as the message's creation time.
+a|
+[.specitem,oft-sid="req~up-attributes-id~1",oft-needs="impl,utest"]
```

Comment:
I wonder if this really is the right place to specify this requirement. It seems quite unclear, where and at what time this would need to be verified. Maybe it would be better to define this in the Transport layer API spec? There we could require that all messages that are sent via a transport must have valid attributes according to the UAttributes spec and we could also define what needs to be done with messages received that are not compliant, e.g. ignore them.

I guess that would also reduce the number of spec items significantly and make it much easier to implement and test the requirements.

WDYT?

Reply:
Sure, but may I at least suggest to turn them into design spec items (using the `dsn` type instead of `req`) as these are not really user level requirements but are the outcome of having thought about how to make it possible ...

Reply:
I guess it depends on what your definitions of feature vs user vs design vs architecture requirements are.  I got the following response from chatGPT:

1. *Feature Requirements:* These are specific functionalities or capabilities that a product or system should have. Feature requirements describe what the product should do or the tasks it should perform to meet user needs or business objectives. They focus on the desired outcomes or features that users expect from the product.

2. *User Requirements:* User requirements capture the needs, expectations, and goals of the end-users or stakeholders who will interact with the product. They define the user's perspective, including their desired functionalities, usability, user experience, and any specific constraints or preferences they may have.

3. *Design Requirements:* Design requirements focus on the visual and interactive aspects of the product. They specify how the product should look, feel, and behave from a user interface (UI) and user experience (UX) perspective. Design requirements may include guidelines for layout, color schemes, typography, navigation, and other visual elements that contribute to the overall design aesthetics and usability.

4. *Architecture Requirements:* Architecture requirements pertain to the underlying structure, components, and technical aspects of the product's system architecture. They define the high-level design principles, performance criteria, scalability, security, integration requirements, and other technical considerations necessary for the product's development and deployment. Architecture requirements guide the development team in designing a robust and scalable system that meets the desired technical specifications.

If we stick with design meaning UI/UX above, we should change the various requirements to `arch` . 

Having said that we do need to agree on definitions of feature vs user vs design vs architecture, my vote is using the definitions above.

Reply:
OFT uses [this definition](https://github.com/itsallcode/openfasttrace/blob/main/doc/user_guide.md#specification-item-artifact-type) where, according to my understanding, `dsn` does not refer to UX requirements ...

Reply:
done